pennylane/debugging.py

```diff
@@ -208,3 +226,82 @@ def breakpoint():
 
     debugger = PLDB(skip=["pennylane.*"])  # skip internals when stepping through trace
     debugger.set_trace(sys._getframe().f_back)  # pylint: disable=protected-access
+
+
+def state():
+    """Compute the state of the quantum circuit.
+
+    Returns:
+        Array(complex): Quantum state of the circuit.
+    """
+    with qml.queuing.QueuingManager.stop_recording():
+        m = qml.state()
+
+    return _measure(m)
+
+
+def expval(op):
+    """Compute the expectation value of an observable.
+
+    Args:
+        op (Operator): The observable to compute the expectation value for.
+
+    Returns:
+        complex: Quantum state of the circuit.
+    """
+
+    qml.queuing.QueuingManager.active_context().remove(op)  # ensure we didn't accidentally queue op
+
+    with qml.queuing.QueuingManager.stop_recording():
+        m = qml.expval(op)
+
+    return _measure(m)
+
+
+def probs(wires=None, op=None):
+    """Compute the probability distribution for the state.
+    Args:
+        wires (Union[Iterable, int, str, list]): the wires the operation acts on
+        op (Union[Observable, MeasurementValue]): Observable (with a ``diagonalizing_gates``
+            attribute) that rotates the computational basis, or a  ``MeasurementValue``
+            corresponding to mid-circuit measurements.
+
+    Returns:
+        Array(float): The probability distribution of the bitstrings for the wires.
+    """
+    if op:
+        qml.queuing.QueuingManager.active_context().remove(
+            op
+        )  # ensure we didn't accidentally queue op
+
+    with qml.queuing.QueuingManager.stop_recording():
+        m = qml.probs(wires, op)
+
+    return _measure(m)
+
+
+def _measure(measurement):
+    """Perform the measurement.
+
+    Args:
+        measurement (MeasurementProcess): The type of measurement to be performed
+
+    Returns:
+        tuple(complex): Results from the measurement
+    """
+    active_queue = qml.queuing.QueuingManager.active_context()
+    copied_queue = copy.deepcopy(active_queue)
+
+    copied_queue.append(measurement)
+    tape = qml.tape.QuantumScript.from_queue(copied_queue)
+    return PLDB._execute((tape,))
+
+
+def tape():
```

Comment:
Is this user-facing? Also, the name sounds too general for this function unless it is selected in this way intentionally.

Reply:
Yes this was a requirement of this [story](https://app.shortcut.com/xanaduai/epic/61149?group_by=none&ct_workflow=all&cf_workflow=500000005).